### DISCUSSION TITLE
GSoC 2024 applications and social networks

### DISCUSSION THREAD

**Discussion by jmplaza** [JdeRobot/jderobot.github.io#242] [General]
If you want to apply to become a GSoC contributor (in case of JdeRobot being accepted again for GSoC this year) it is good to:

1. start soon using our software (RoboticsAcademy, VisualCircuit, BehaviorMetrics...) .
2. interact with JdeRobot community for questions or sharing achievements (a successful installation, examples using our software...) through our repositories and Discussion channels.
3. show your achievements in the social networks, for instance in Twitter (X), in YouTube and LinkedIn.... (a short video, a snapshot...) Just mention there JdeRobot account so we will get notifiied.
4. contribute with some Pull Requests.

This year our candidate selection process will take into account the previous interactions along the last months. 
Cheers!

@Samkhan705279 @zeeshan8302 @aniketsona23 @AnushkaGharage @obamwonyi @BkPankaj

**Comment by obamwonyi** [comment]
I have setup Robotics Academy and I am currently learning with it , I am currently having issues setting up visual circuit , but I believe I will fix it soon enough .

**Reply by BkPankaj** [reply]
Hello @obamwonyi, If you have any issues that are not being solved, feel free to contact me, as I have also encountered several issues during the installation of Visual Circuit. Additionally, I am experiencing some difficulties in RoboticsAcademy. If you have completed the setup, could you please provide the solution or some hints? You can find the discussion regarding the RoboticsAcademy issue [here](https://github.com/JdeRobot/RoboticsAcademy/discussions/2387#discussioncomment-8374252).

**Comment by SohamPatel6395** [comment]
Hello, I have been trying to complete some of the challenges for GSoC 2024 for JdeRobot. For the Robotics Academy Challenge, I have downloaded and installed Docker, but it keeps crashing when I try to open Docker. I currently use a Mac m1 chip. For now, I am just gonna do the next challenge, but I would like to know if anyone has any suggestions on how to fix this.

**Reply by jmplaza** [reply]
Hi @SohamPatel6395, 

we will update the programming tests soon. 

Regarding the use of RADI with RoboticsAcademy, take a look at [this thread](https://github.com/orgs/JdeRobot/discussions/239#discussioncomment-8243895) and [this answer with promising links to test](https://github.com/orgs/JdeRobot/discussions/239#discussioncomment-8266221).

**Reply by obamwonyi** [reply]
Hi @jmplaza  I know from your contribution guide  exercise are to be implemented in ROS nuetic or Foxy (ROS2), but this version are no longer supported , and even ros documentation discourage installing and setting any of this distributions up , I currently have iron . Is it ok If I use Iron?,  or would you prefer I create a vm or dual boot for 20.04 as my distro 22.04 does not have a release for nuetic or Foxy .

**Reply by SohamPatel6395** [reply]
Oh sorry, I didn't notice that the programming tests aren't updated yet. I already completed the C++ challenge and Python by mistake. I will look into the threads you provided. Thanks.

**Reply by jmplaza** [reply]
Hi @obamwonyi , the contribution guide is somehow outdated. The current supported ROS releases in RoboticsAcademy are ROS1 Noetic and mainly ROS2 Humble. The last one is the preferred one. Cheers.

**Comment by AryanNanda17** [comment]
> Hello, I have been trying to complete some of the challenges for GSoC 2024 for JdeRobot. For the Robotics Academy Challenge, I have downloaded and installed Docker, but it keeps crashing when I try to open Docker. I currently use a Mac m1 chip. For now, I am just gonna do the next challenge, but I would like to know if anyone has any suggestions on how to fix this.

Hello @SohamPatel6395, I faced the same issue during my initial stage of Jde. You have to run docker using rosetta emulation and then the performance would be much better. Let me know if you face more issues.

**Comment by BkPankaj** [comment]
Greeting everyone,

I have a doubt about the line highlighted in the image below (i.e., Project 5 VisualCircuit block library).
![](https://github.com/JdeRobot/jderobot.github.io/assets/62588059/779f4cf2-742c-4897-b607-b24428833dfd)

Will the automated testing be conducted manually by maintainers before accepting the block for the repository? Or, will there be automated testing where users provide inputs and outputs, and the system automatically verifies and accepts the block for the repository? Can someone clarify this aspect and provide further details?

**Reply by jmplaza** [reply]
Hi @BkPankaj, 

it is yet to be explored and defined. The main goal is to introduce some automated tests into VisualCircuit, at several levels, to improve its software quality and adding new contributions without breaking any functionality.

For instance, tests for VisualCircuit tool itself, such as Selenium based ones for visual editor, tests for the translator to Python code, and even tests for the global execution. In addition for each new block maybe we should require each contributor to add the corresponding unit tests for that block....

**Reply by BkPankaj** [reply]
Thank you @jmplaza  for clearing up the doubt. I sincerely appreciate your guidance. I will take your points into consideration and ensure they are reflected in the proposal.

**Comment by Chijiaku** [comment]
Greetings everyone,

This is my first time in applying to GSoc 2024.  I have a very keen interest in  actively contributing to one of the open source projects for a long period of my career in this organization JdeRobot.
 is it late for me to contribute and participate in the challenges seeing I have a week deadline to do so?
also, am I to take the 4 challenges together?
I have started by already though.


I'll appreciate the responses to these questions.

Thanks,
Emmanuel.

**Reply by jmplaza** [reply]
The deadline is April 2nd, you have some days ahead :-) A little bit late to receive feedback about your official application from potential mentors, to refine it, anyway.

You have to do only the challenges related to the project(s) of https://jderobot.github.io/activities/gsoc/2024 you are interested in. They are indicated on the [application instructions](https://jderobot.github.io/activities/gsoc/2024#application-instructions-for-gsoc-2024).

Cheers,

**Reply by Chijiaku** [reply]
Thank you for your response sir.

I can still do something before 2nd April so much hours before then. 
I didn't know about the official application. Is there a form or something I have to fill for the application? or I just have to go on with my interest on the challenges?

Thanks.

**Reply by jmplaza** [reply]
Hi @cmanny1, 

just read the documentation and instructions, both at the [official GSoC webpage](https://summerofcode.withgoogle.com/) and at [our org's GSoC-2024 webpage](https://jderobot.github.io/activities/gsoc/2024).

Cheers!

**Reply by Chijiaku** [reply]
Thank you.

**Reply by Chijiaku** [reply]
Hi @jmplaza , I have completed a challenge relating to the projects I am interested in and applied with the google form I saw at JdeRobot webpage.

Thanks.

**Comment by Chijiaku** [comment]
Greetings @jmplaza,
what do I need to do next? I am interested in project 4 and project 10.
I've got idea on how to implement them.
how do I get about the mentor reaching out so I can write my proposal before the 2nd of April. 

I'll appreciate the responses.

Thanks,
Emmanuel.

**Reply by jmplaza** [reply]
Hi @cmanny1 ,

just read the documentation and instructions, both at the official GSoC webpage and at [our org's GSoC-2024 webpage.
](https://jderobot.github.io/activities/gsoc/2024#application-instructions-for-gsoc-2024). For instance, in the JdeRobot GSoC-2024 web page: _AFTER doing the programming tests, fill this web form with your information and challenge results. Then you are invited to ask the project mentors about the project details. _ 

You have the email addresses of the potential mentors on that web page too, to interact with them, and the link to fill that web form. 

In parallel you have to upload your final application to the official GSoC webpage. Submit it even if you do not receive any feedback from the potential mentors. As I said, now it is kind of late to interact with them, and we do our best. 

Cheers,
